src/widget/wsearchlineedit.cpp

```diff
@@ -387,10 +387,17 @@ void WSearchLineEdit::keyPressEvent(QKeyEvent* keyEvent) {
         }
         break;
     case Qt::Key_Left:
-    case Qt::Key_Right:
+    case Qt::Key_Right: {
+        QLineEdit* pEdit = lineEdit();
+        bool hadSelectedText = pEdit->hasSelectedText();
         QComboBox::keyPressEvent(keyEvent);
```

Comment:
How could you be sure that the pointer returned by lineEdit() remains valid after changing the state of the object instance by invoking QComboBox::keyPressEvent()? There is no such promise.